### PR TITLE
Adding new syscalls

### PR DESCRIPTION
**Pull request by geoffret**
Added syscalls for checking if current tasks' budget has exhausted, and to check slack time until deadline.  

I sent a more detailed email to the mailing list. 

I did some basic testing on this with the liblitmus patch, but nothing too in detail.

### CODE REVIEW

**Comment by geoffret** [comment]
Please see discussion at LITMUS-RT/liblitmus/#1

File: litmus/litmus.c [hunk 1]
```diff
@@ -314,6 +316,50 @@ asmlinkage long sys_null_call(cycles_t __user *ts)
...
+ *
+ *	returns 0 on successful query
+ *	returns EPERM if task is not a real-time task.
+ *      returns EFAULT if &job is not a valid pointer.
+ */
+asmlinkage long sys_query_budget_exhausted(int __user *flag)
+{
+	int retval = -EPERM;
+	if (is_realtime(current))
+		retval = put_user(budget_exhausted(current), flag);
```
Comment (brandenburg) [inline litmus/litmus.c]: `budget_exhausted()` relies on the scheduler's time-tracking services. In particular, in order for `budget_exhausted()` to be correct, the scheduler must have "recently" charged execution costs. This, however, is done only prior to a call to `schedule()`, unless it is explicitly triggered. I'm afraid this patch works superficially, but reports wrong values if you have jobs that run without preemption for a couple of milliseconds.

File: litmus/litmus.c [hunk 2]
```diff
@@ -314,6 +316,50 @@ asmlinkage long sys_null_call(cycles_t __user *ts)
...
+	int retval = -EPERM;
+	lt_t currTime = litmus_clock();
+	lt_t deadline = current->rt_param.job_params.deadline;
+
+	if (is_realtime(current))
+	{
+		// retval = put_user(budget_remaining(current), exec_time);
+		if(currTime < deadline)
+		{
+			retval = put_user(deadline - currTime, exec_time);
```
Comment (brandenburg) [inline litmus/litmus.c]: As Glenn noted, it would be better to do this through the control page.

Comment (brandenburg) [inline litmus/litmus.c]: Also, if you need precise control over deadlines, you may want to use `clock_nanosleep()` to trigger job wake-ups.

**Comment by geoffret** [comment]
Changed the interface to get the job deadlines to use control pages, as recommended at LITMUS-RT/liblitmus#1.

Closing this pull request, as the implementation comes from a different branch.

**Pull request closed without merging**